### ISSUE TITLE
compile docker source code related issue

### ISSUE THREAD

**Issue by michael4RD** [docker/docker#15761]
Hi,

I am new to docker and go. I may have a naive question on compiling from docker source code: I could not find instructions on how to build docker from its source code. I did some research online, and followed these instructions "cd docker/docker && go build && ./docker version", but go complains about file not found, GOPATH/GOROOT not set (set actually). I also tried to "make" from the root directory of the source code, but it failed as well complaining error retrieving images. I am using the latest docker source code, and the latest go.

So, simply put, could someone guide me how to compile docker from its source code instead of wget the binaries?

Thanks for your time. Appreciate it!

Best,

**Comment by cpuguy83** [comment]
Here's a guide on our docs page: https://docs.docker.com/project/set-up-dev-env/

Generally though, with docker already running, in the root of the docker git repo, type `make BINDDIR=. binary`
Feel free to ping us on #docker-dev on IRC for help.
Thanks!

**Comment by michael4RD** [comment]
Problem solved. It seems the company network restriction is the reason for building failures :)
